tests/test_expression.py

```diff
@@ -118,7 +118,12 @@ def get_context_data(
 
         self.assertEqual(
             rendered.strip(),
-            "<!-- _RENDERED SimpleComponent_5b8d97,a1bc3f,, -->\n<div data-djc-id-a1bc3f>lorem</div>\n<div data-djc-id-a1bc3f>True</div>\n<div data-djc-id-a1bc3f>[{'a': 1}, {'a': 2}]</div>",  # noqa: E501
+            (
+                "<!-- _RENDERED SimpleComponent_5b8d97,a1bc3f,, -->\n"
```

Comment:
BS4 stripped extra whitespace, while our custom pure Python implementation leaves it as is, hence these changes.

Reply:
I think fixing the tests would be a more long-term bet in this case. We don't care about the whitespace, so lets skip it in the tests.

Reply:
Updated in https://github.com/django-components/django-components/pull/927/commits/e756713adc78fd41dba35b808ae2e90af67f3943.

There's 2-3 tests that are whitespace-sensitive, so there I've kept using `assertEqual`. The rest of the tests I converted to use `assertHTMLEqual`, which ignores whitespace.